plugins/jenkins-backend/README.md

```diff
@@ -0,0 +1,204 @@
+# Jenkins Plugin (Alpha)
+
+Welcome to the Jenkins backend plugin! Website: [https://jenkins.io/](https://jenkins.io/)
+
+This is the backend half of the 2 Jenkins plugins and is responsible for:
+
+- finding an appropriate instance of Jenkins for an entity
+- finding the appropriate job(s) on that instance for an entity
+- connecting to Jenkins and gathering data to present to the frontend
+
+## Integrating into a backstage instance
+
+This plugin needs to be added to an existing backstage instance.
+
+Typically, this means creating a `src/plugins/jenkins.ts` file and adding a reference to it to `src/index.ts`
+
+### jenkins.ts
+
+```typescript
+import {
+  createRouter,
+  DefaultJenkinsInfoProvider,
+} from '@backstage/plugin-jenkins-backend';
+import { CatalogClient } from '@backstage/catalog-client';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
+
+export default async function createPlugin({
+  logger,
+  config,
+  discovery,
+}: PluginEnvironment): Promise<Router> {
+  const catalogClient = new CatalogClient({ discoveryApi: discovery });
+
+  return await createRouter({
+    logger,
+    jenkinsInfoProvider: new DefaultJenkinsInfoProvider(catalogClient, config),
+  });
+}
+```
+
+This plugin must be provided with a JenkinsInfoProvider, this is a strategy object for finding the Jenkins instance and job for an entity.
+
+There is a standard one provided, but the Integrator is free to build their own.
+
+### DefaultJenkinsInfoProvider
+
+Allows configuration of either a single or multiple global Jenkins instances and annotating entities with the job name on that instance (and optionally the name of the instance).
+
+#### Example - Single global instance
+
+The following will look for jobs for this entity at `https://jenkins.example.com/job/teamA/job/artistLookup-build`
+
+Config
+
+```yaml
+jenkins:
+  baseUrl: https://jenkins.example.com
+  username: backstage-bot
+  apiKey: 123456789abcdef0123456789abcedf012
+```
+
+Catalog
+
+```yaml
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: artist-lookup
+  annotations:
+    'jenkins.io/job-slug': teamA/artistLookup-build
+```
+
+The old annotation name of `jenkins.io/github-folder` is equivalent to `jenkins.io/job-slug`
+
+#### Example - Multiple global instances
+
+The following will look for jobs for this entity at `https://jenkins-foo.example.com/job/teamA/job/artistLookup-build`
+
+Config
+
+```yaml
+jenkins:
+  instances:
+    - name: default
+      baseUrl: https://jenkins.example.com
+      username: backstage-bot
+      apiKey: 123456789abcdef0123456789abcedf012
+    - name: departmentFoo
+      baseUrl: https://jenkins-foo.example.com
+      username: backstage-bot
+      apiKey: 123456789abcdef0123456789abcedf012
+```
+
+Catalog
+
+```yaml
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: artist-lookup
+  annotations:
+    'jenkins.io/job-slug': departmentFoo:teamA/artistLookup-build
```

Comment:
Here you are using `job-slug` and in other parts you are mentioning `folder-slug`.

Reply:
good spot. I was back and forward and obviously missed the docs once I had decided :-P